pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/util/UserUtil.java

```diff
@@ -268,6 +271,29 @@ public static boolean checkUserAccess(String realm, String userId, String role,
         return false;
     }
 
+    public static boolean checkUserAccessForPatch(String realm, String userId, String role, String userIdRequest, Patch patch) {
```

Comment:
Can you comment on what we are doing here?
It looks like we are getting an email or ID from the request body but then check that is it also the verified user ID?(We also do this in checkUserAccess, but I don't understand what that accomplishes. Is it because Patch uses the user ID/email from the content at some point?)
That will help me better understand your tests.

Reply:
Of course!
The `userId` that is passed as an input is the userId that is retrieved from the token that was passed in this call. However, the `patch` object also contains a userId in it, which we use as an audit field to know who changed this record in the last edit. This method is supposed to check that what is passed in the `patch` object matches the token.

Patch calls from the frontend are inconsistent, some include the user emails in the `userId` and some the int value of the userId. We wanted to move to use the emails but it is not consistent, and I wanted to make sure we are covering both cases.

The `checkUserAccess` is matching what was passed in the request to what was in the token. So basically in patch calls 3 user identifiers are being passed to the backend, and we want to make sure they all match.

Reply:
Thx!
It seems the main use for the ID in the patch is for auditing.